code-studio/src/js/components/progress/progress_dot.jsx

```diff
@@ -132,36 +168,32 @@ const ProgressDot = React.createClass({
       onClick = (e) => {this.dotClicked(level.url); e.preventDefault();};
     }
 
-    if (level.kind === 'named_level') {
-      let icon = <i className={`fa ${level.icon}`} style={dotStyle} />;
-      if (level.show_progress) {
-        icon = <span style={dotStyle}>&nbsp;</span>;
-        Object.assign(dotStyle, styles.status[level.status || 'not_tried']);
-      } else {
-        Object.assign(dotStyle, styles.dot.without_progress);
+    let dot, name, outerStyle;
+    if (!level.icon) {
+      Object.assign(dotStyle, styles.status[level.status || 'not_tried']);
+      let dotText = level.title;
+      if (level.kind === 'named_level') {
+        // Non-breaking space (&nbsp;)
+        dotText = '\u00a0';
       }
+      dot = <div style={dotStyle} className={`level-${level.id}`}>{dotText}</div>;
```

Comment:
I feel like you can collapse the above into a ternary here - {level.kind === 'named_level' ? &nbsp : level.title}